target/linux/mediatek/mt7623/base-files/lib/preinit/07_set_iface_mac

```diff
@@ -29,17 +21,37 @@ preinit_set_mac_address() {
 
 			sync
 			umount /tmp/recovery
-			rm -rf /tmp/recovery
+			rmdir /tmp/recovery
+
+			ip link set dev wan address $mac 2> /dev/null
+
+			mac=$(macaddr_add $mac 1)
+
+			ip link set dev lan0 address $mac 2>/dev/null
+			ip link set dev lan1 address $mac 2>/dev/null
+			ip link set dev lan2 address $mac 2>/dev/null
+			ip link set dev lan3 address $mac 2>/dev/null
 		fi
+	fi
+}
 
-		ip link set dev wan address $mac 2> /dev/null
+preinit_set_mac_address() {
 
-		mac=$(macaddr_add $mac 1)
+	. /lib/functions.sh
```

Comment:
Could you move the include lines to outside of function? [ref](https://github.com/openwrt/openwrt/blob/master/target/linux/mvebu/base-files/lib/preinit/79_move_config#L4)